source/dagger/gameplay/team_game/tilemap_legends.cpp

```diff
@@ -0,0 +1,121 @@
+	Entity entity = reg_.create();
+	auto& sprite = reg_.emplace<Sprite>(entity);
+	sprite.scale = { scale,scale };
+	sprite.position = { x_ * x_step, y_ * y_step, 1000 };
+	AssignSprite(sprite, "spritesheets:tiles_dungeon:floor_1");
+	return entity;
+}
+Entity CreateWall(Registry& reg_, UInt32 x_, UInt32 y_, char type) {
+	Map<Char, String>walls;
+	Entity entity = reg_.create();
+	auto& sprite = reg_.emplace<Sprite>(entity);
+	sprite.scale = { scale,scale };
+	sprite.position = { x_ * x_step, y_ * y_step, 10 };
+	AssignSprite(sprite, "spritesheets:tiles_dungeon:"+wall_type.at(type));
+	return entity;
+}
+Entity CreateSlime(Registry& reg_, UInt32 x_, UInt32 y_, char type) {
+	Entity entity = reg_.create();
+	auto& sprite = reg_.emplace<Sprite>(entity);
+	sprite.scale = { scale,scale };
+	sprite.position = { x_ * x_step, y_ * y_step, 30 };
+	AssignSprite(sprite, "spritesheets:chara_slime:slime_idle_anim:1");
+	auto& anim = reg_.emplace<Animator>(entity);
+	AnimatorPlay(anim, "chara_slime:slime_idle");
+	return entity;
+}
+Entity CreateHero(Registry& reg_, UInt32 x_, UInt32 y_, char type) {
+	auto view = Engine::Registry().view<TeamGameCharacter,Sprite>();
+	auto it = view.begin();
+	auto& sprite = view.get<Sprite>(*it);
+	sprite.position = { x_ * x_step,y_ * y_step,sprite.position.z };
+	return *it;
+}
+Entity CreateDoorHorizontal(Registry& reg_, UInt32 x_, UInt32 y_, char type) {
+	Entity entity = reg_.create();
+	auto& sprite = reg_.emplace<Sprite>(entity);
+	sprite.scale = { scale,scale };
+	sprite.position = { x_ * x_step, y_ * y_step, 40 };
+	AssignSprite(sprite, "spritesheets:tiles_dungeon:door_horizontal");
+	return entity;
+}
+Entity CreateDoorVertical(Registry& reg_, UInt32 x_, UInt32 y_, char type) {
+	Entity entity = reg_.create();
+	auto& sprite = reg_.emplace<Sprite>(entity);
+	sprite.scale = { scale,scale };
+	sprite.position = { x_ * x_step, y_ * y_step, 40 };
+	AssignSprite(sprite, "spritesheets:tiles_dungeon:door_vertical");
+	return entity;
+}
+Entity CreateChest(Registry& reg_, UInt32 x_, UInt32 y_, char type) {
+	Entity entity = reg_.create();
+	auto& sprite = reg_.emplace<Sprite>(entity);
+	sprite.scale = { scale,scale };
+	sprite.position = { x_ * x_step, y_ * y_step, 90 };
+	AssignSprite(sprite, "spritesheets:tiles_dungeon:chest");
+	return entity;
+}
+Entity CreatePot(Registry& reg_, UInt32 x_, UInt32 y_, char type) {
+	Entity entity = reg_.create();
+	auto& sprite = reg_.emplace<Sprite>(entity);
+	sprite.scale = { scale,scale };
+	sprite.position = { x_ * x_step, y_ * y_step, 90 };
+	AssignSprite(sprite, "spritesheets:tiles_dungeon:pot");
+	return entity;
+}
+Entity CreateCrate(Registry& reg_, UInt32 x_, UInt32 y_, char type) {
+	Entity entity = reg_.create();
+	auto& sprite = reg_.emplace<Sprite>(entity);
+	sprite.scale = { scale,scale };
+	sprite.position = { x_ * x_step, y_ * y_step, 90 };
+	AssignSprite(sprite, "spritesheets:tiles_dungeon:crate");
+	return entity;
+}
```

Comment:
There are many problems here but I want team members to review. Do not merge this as-is.